### PR TITLE
Support param for field from Oneof definition.

### PR DESCRIPTION
**Pull request by bonafideyan**
when the param binds with a field defined by Oneof, it can not be assigned directly.

### CODE REVIEW

File: examples/proto/examplepb/echo_service.proto [hunk 1]
```diff
@@ -9,11 +9,29 @@ package grpc.gateway.examples.examplepb;
...
 // SimpleMessage represents a simple message sent to the Echo service.
 message SimpleMessage {
 	// Id represents the message identifier.
 	string id = 1;
 	int64 num = 2;
+	oneof code {
+		int64 line_num = 3;
+		string lang = 4;
+	}
+	int64 time = 5;
```
Comment (yugui) [inline examples/proto/examplepb/echo_service.proto]: Can you remove `time` if you are not using it?
I'd like to keep `SimpleMessage` as simple as possible.

Comment (bonafideyan) [reply examples/proto/examplepb/echo_service.proto]: Done.

File: examples/integration/integration_test.go [hunk 2]
```diff
@@ -98,6 +101,105 @@ func testEcho(t *testing.T, port int, contentType string) {
...
+		t.Logf("%s", buf)
+	}
+
+	var msg gw.SimpleMessage
+	if err := jsonpb.UnmarshalString(string(buf), &msg); err != nil {
+		t.Errorf("jsonpb.UnmarshalString(%s, &msg) failed with %v; want success", buf, err)
+		return
+	}
+	if got, want := msg.GetNo().GetNote(), "golang"; got != want {
+		t.Errorf("msg.Lang = %q; want %q", got, want)
```
[the remark this answers is not in the document]
Comment (bonafideyan) [reply examples/integration/integration_test.go]: Done.

File: examples/integration/integration_test.go [hunk 3]
```diff
@@ -98,6 +101,105 @@ func testEcho(t *testing.T, port int, contentType string) {
...
+		t.Logf("%s", buf)
+	}
+
+	var msg gw.SimpleMessage
+	if err := jsonpb.UnmarshalString(string(buf), &msg); err != nil {
+		t.Errorf("jsonpb.UnmarshalString(%s, &msg) failed with %v; want success", buf, err)
+		return
+	}
+	if got, want := msg.GetStatus().GetNote(), "golang"; got != want {
+		t.Errorf("msg.Lang = %q; want %q", got, want)
```
[the remark this answers is not in the document]
Comment (bonafideyan) [reply examples/integration/integration_test.go]: Done.

**Comment by bonafideyan** [comment]
I didn't know those files need be regenerated and reason of build failure.
Fixed, thank you.

**Comment by yugui** [comment]
LGTM. Thank you for your contribution.

**Pull request merged**